HLSMergerPlugin/src/main/java/io/antmedia/plugin/HLSMergerPlugin.java

```diff
@@ -0,0 +1,181 @@
+						.append(programId)
+						.append(",BANDWIDTH=").append(bitrate)
+						.append(",RESOLUTION=").append(width).append("x").append(height);
+				if (codec == AV_CODEC_ID_H264) {
+					streamIdBuilder.append(",CODECS=\"avc1.42e00a,mp4a.40.2\"");
+				}
+				else if (codec == AV_CODEC_ID_H265) {
+					streamIdBuilder.append(",CODECS=\"hvc1.1.4.L60.B01,mp4a.40.2\"");
+				}
+				streamIdBuilder.append("\n");
+
+				streamIdBuilder.append(streamId + ".m3u8")
+						.append("\n");
+			}
+		}
+
+		writeHLSFile(fileName, streamIdBuilder.toString(), subfolder);
+    }
+
+	public boolean stopMerge(String fileName) {
+		if (mergeTimers.containsKey(fileName)) {
+			long hlsMergedFileUpdate = mergeTimers.get(fileName);
+			vertx.cancelTimer(hlsMergedFileUpdate);
+			mergeTimers.remove(fileName);
+		}
+		return true;
+	}
+
+	public boolean addMultipleAudioStreams(String fileName, String videoStreamId, String[] audioStreamIds) {
+		long hlsMergedFileUpdate = vertx.setPeriodic(5000,h -> addMultipleAudioInternal(fileName, videoStreamId, audioStreamIds));
+		mergeTimers.put(fileName, hlsMergedFileUpdate);
+		return true;
+	}
+	
+	public void addMultipleAudioInternal(String fileName, String videoStreamId, String[] audioStreamIds) {
+		StringBuilder streamBuilder = new StringBuilder();
+		String subfolder = "";
+		streamBuilder.append("#EXTM3U\n");
+
+		// Add audio streams
+		for (String audioStreamId : audioStreamIds) {
+			MuxAdaptor muxAdaptor = getMuxAdaptor(audioStreamId);
+			if (muxAdaptor != null) {
+				subfolder = muxAdaptor.getBroadcast().getSubFolder();
+				streamBuilder.append("#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"audio\",NAME=\"")
+					.append(audioStreamId)
+					.append("\",DEFAULT=NO,AUTOSELECT=YES,URI=\"")
+					.append(audioStreamId).append(".m3u8\"\n");
+			}
+		}
+
+		// Add video stream with reference to audio group
+		MuxAdaptor videoMuxAdaptor = getMuxAdaptor(videoStreamId);
+		if (videoMuxAdaptor != null) {
+			long bitrate = 0;
+			for (Muxer muxer : videoMuxAdaptor.getMuxerList()) {
+				if (muxer instanceof HLSMuxer) {
+					HLSMuxer hlsMuxer = (HLSMuxer) muxer;
+					bitrate = hlsMuxer.getAverageBitrate();
+				}
+			}
+			int width = videoMuxAdaptor.getWidth();
+			int height = videoMuxAdaptor.getHeight();
+			streamBuilder.append("#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=")
+				.append(bitrate)
+				.append(",RESOLUTION=").append(width).append("x").append(height)
+				.append(",CODECS=\"avc1.42e00a,mp4a.40.2\",AUDIO=\"audio\"\n");
+			streamBuilder.append(videoStreamId).append(".m3u8\n");
+		}
+
+		writeHLSFile(fileName, streamBuilder.toString(), subfolder);
+	}
+
+	public void writeHLSFile(String fileName, String content, String subfolder) {
+		IScope scope = getApplication().getScope();
+		File audioHLSFile = Muxer.getRecordFile(scope, fileName, ".m3u8.tmp", subfolder);
+
+		try (PrintWriter out = new PrintWriter(audioHLSFile)) {
+			out.println(content);
+		} catch (FileNotFoundException e) {
```

Comment:
Please merge the exceptions because if it fails here,  it does not make sense to run the codes below